src/layers.c

```diff
@@ -118,45 +118,116 @@ handle_output_destroy(struct wl_listener *listener, void *data)
 	wlr_layer_surface_v1_destroy(layer->scene_layer_surface->layer_surface);
 }
 
-static void
-process_keyboard_interactivity(struct lab_layer_surface *layer)
+static inline bool
+has_exclusive_interactivity(struct wlr_scene_layer_surface_v1 *scene)
 {
-	struct wlr_layer_surface_v1 *layer_surface = layer->scene_layer_surface->layer_surface;
-	struct seat *seat = &layer->server->seat;
+	return scene->layer_surface->current.keyboard_interactive
+		== ZWLR_LAYER_SURFACE_V1_KEYBOARD_INTERACTIVITY_EXCLUSIVE;
+}
 
-	if (layer_surface->current.keyboard_interactive
-			&& layer_surface->current.layer >= ZWLR_LAYER_SHELL_V1_LAYER_TOP) {
+/*
+ * Try to transfer focus other layer-shell clients with exclusive focus on the
+ * output nearest to the cursor. In none exist (which is likely to generally be
+ * the case) just unset layer focus and try to give it to the topmost toplevel
+ * if one exists.
+ */
+static void
+try_to_focus_next_layer_or_toplevel(struct server *server)
+{
+	struct seat *seat = &server->seat;
+	struct output *output = output_nearest_to_cursor(server);
+
+	int overlay = ZWLR_LAYER_SHELL_V1_LAYER_OVERLAY;
+	int top = ZWLR_LAYER_SHELL_V1_LAYER_TOP;
```

Comment:
Should we `s/int/enum zwlr_layer_shell_v1_layer/` here?

Reply:
Yes, that's better. Done.